content-scripts/src/modules/schedule.js

```diff
@@ -165,15 +166,7 @@ const fixClasses = (table) => {
 };
 
 const getClassDuration = async (url) => {
-    const r = await fetch(url);
-
-    const decoder = new TextDecoder(
-        r.headers.get("Content-Type").replace("text/html; charset=", "")
-    );
-    const text = decoder.decode(await r.arrayBuffer());
-
-    const parser = new DOMParser();
-    let html = parser.parseFromString(text, "text/html");
+    const html = await fetchSigarraPage(url);
```

Comment:
This breaks the code further down, as the `decoder` and `parser` variables are no longer in scope. An easy way to solve this is to replace the code inside the if statement with a recursive call to this same function. i.e:

```diff
  if (a) {
-     const r = await fetch(a.href);
-     const text = decoder.decode(await r.arrayBuffer());
-     html = parser.parseFromString(text, "text/html");
+     return await getClassDuration(a.href);
  }
```